controllers/secret_controller.go

```diff
@@ -163,35 +163,46 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		return reconcile.Result{}, nil
 	} else {
 		// userdata secret data does not match what is expected
-		// Mark nodes configured with the previous private key for deletion
 		nodes := &core.NodeList{}
 		err = r.client.List(ctx, nodes, client.MatchingLabels{core.LabelOSStable: "windows"})
 		if err != nil {
 			return reconcile.Result{}, errors.Wrapf(err, "error getting node list")
 		}
+
+		// Modify annotations on nodes configured with the previous private key, if it has changed
 		expectedPubKeyAnno := nodeconfig.CreatePubKeyHashAnnotation(keySigner.PublicKey())
-		patchData, err := metadata.GenerateAddPatch(nil, map[string]string{nodeconfig.PubKeyHashAnnotation: ""})
+		privateKeyBytes, err := secrets.GetPrivateKey(kubeTypes.NamespacedName{Namespace: r.watchNamespace,
+			Name: secrets.PrivateKeySecret}, r.client)
 		if err != nil {
-			return reconcile.Result{}, errors.Wrapf(err, "error creating public key annotation add request")
+			return reconcile.Result{}, err
 		}
 		for _, node := range nodes.Items {
-			// Only clear the annotation for Nodes associated with Machines, as the clearing of the annotation is used
-			// solely to kick off the deletion and recreation of Machines.
-			if _, present := node.Annotations[BYOHAnnotation]; present {
+			if node.Annotations[nodeconfig.PubKeyHashAnnotation] == expectedPubKeyAnno {
```

Comment:
Yep, these 2 annotations are [applied at the same time](https://github.com/openshift/windows-machine-config-operator/blob/2ceed3d84b140aac958071c34e812a39a2efbde1/pkg/nodeconfig/nodeconfig.go#L177-L178) and updated together if needed. Since both are dependent on the private key secret, I concluded the username check was therefore redundant

Reply:
I suggest adding a comment stating that here.